gulpfile.js

```
@@ -297,6 +303,216 @@ gulp.task('minifyRelease', ['generateStubs'], function() {
+                // compute hash and etag
+                var hash = crypto.createHash('md5').update(content).digest('hex');
+                etag = crypto.createHash('md5').update(content).digest('base64');
+
+                var index = existingBlobs.indexOf(blobName);
+                if (index <= -1) {
+                    return content;
+                }
+
+                // remove files as we find them on disk
+                existingBlobs.splice(index, 1);
+
+                // get file info
+                return s3.headObjectAsync({
+                    Bucket : bucketName,
+                    Key : blobName
+                })
+                .then(function(data) {
+                    if (data.ETag !== ('"' + hash + '"') ||
+                        data.CacheControl !== cacheControl ||
+                        data.ContentType !== contentType ||
+                        data.ContentEncoding !== contentEncoding) {
+                        return content;
+                    }
+
+                    // We don't need to upload this file again
+                    skipped++;
+                    return undefined;
+                })
+                .catch(function() {
+                    return content;
+                });
+            })
+            .then(function(content) {
+                if (!content) {
+                    return;
+                }
+
+                console.log('Uploading ' + blobName + '...');
+                var params = {
+                    Bucket : bucketName,
+                    Key : blobName,
+                    Body : content,
+                    ContentMD5 : etag,
+                    ContentType : contentType,
+                    ContentEncoding : contentEncoding,
+                    CacheControl : cacheControl
+                };
+
+                return s3.putObjectAsync(params).then(function() {
+                    uploaded++;
+                })
+                .catch(function(error) {
```

This catch should go one level down because if `readFile` or any of the other blocks cause an error, it will be silently ignored.  By moving it to the parent promise chain, it will catch any and all errors.